src/go/rdctl/cmd/factoryReset.go

```
@@ -40,23 +48,264 @@ Use the --remove-kubernetes-cache=BOOLEAN flag to also remove the cached Kuberne
+		appHomePath,
+		altAppHomePath,
+		configPath,
+		logsPath,
+		updaterPath,
+		path.Join(libraryPath, "Application Support", "Rancher Desktop"),
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(altAppHomePath, path.Join(os.Getenv("HOME"), ".config"), pathList)
+}
+
+func deleteLinuxData() error {
+	homeDir := os.Getenv("HOME")
+
+	dataHomePath := os.Getenv("XDG_DATA_HOME")
+	if dataHomePath == "" {
+		dataHomePath = path.Join(homeDir, ".local", "share")
+	}
+	dataHomePath = path.Join(dataHomePath, "rancher-desktop")
+
+	configHomePath := os.Getenv("XDG_CONFIG_HOME")
+	if configHomePath == "" {
+		configHomePath = path.Join(homeDir, ".config")
+	}
+	configPath := path.Join(configHomePath, "rancher-desktop")
+
+	cacheHomePath := os.Getenv("XDG_CACHE_HOME")
+	if cacheHomePath == "" {
+		cacheHomePath = path.Join(homeDir, ".cache")
+	}
+	cachePath := path.Join(cacheHomePath, "rancher-desktop")
+
+	altAppHomePath := path.Join(homeDir, ".rd")
+
+	pathList := []string{
+		altAppHomePath,
+		configPath,
+		path.Join(configHomePath, "Rancher Desktop"),
+		path.Join(homeDir, ".local", "state", "rancher-desktop"),
+	}
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath != "" {
+		pathList = append(pathList, logsPath, path.Join(dataHomePath, "lima"))
+	} else {
+		pathList = append(pathList, path.Join(dataHomePath))
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(altAppHomePath, configHomePath, pathList)
+}
+
+// XXX: Windows doesn't have symlinks, do we have to clean up any cli-plugins?
+func deleteWindowsData() error {
+	err := factoryreset.UnregisterWSL()
+	if err != nil {
+		return err
+	}
+	err = factoryreset.DeleteWindowsData(!removeKubernetesCache, "rancher-desktop")
+	if err != nil {
+		return err
+	}
+	return clearDockerContext()
+}
+
+func deleteLinuxLikeData(altAppHomePath string, configHomePath string, pathList []string) error {
+	deleteLimaVM()
```

We're ignoring the error return here; that seems odd?

---

Most likely because I was transliterating Jan's shell script to this, and it was ignoring problems.